vpn.py

```diff
@@ -64,23 +68,76 @@ def scan_vpns(directory, auth_file):
         # to run
         config = centinel.config.Configuration()
         config.parse_config(centinel_config)
+
+        # assuming that each VPN config file has a name like:
+        # [ip-address].ovpn, we can extract IP address from filename
+        # and use it to geolocate and fetch experiments before connecting
+        # to VPN.
+        vpn_address, extension = os.path.splitext(filename)
+        country = None
+        try:
+            geo = centinel.backend.geolocate(config.params,
```

Comment:
That seems excessively long (perhaps we need to split the files across multiple VMs). In general, providing a side channel about what we are doing makes me a bit uncomfortable when it's not necessary. It doesn't really affect security here, but it seems unnecessary.